bin/zerobufCxx.py

```diff
@@ -330,16 +349,22 @@ def from_json(self):
         if self.value_type.is_zerobuf_type:
             return '::zerobuf::fromJSON( ::zerobuf::getJSONField( json, "{0}" ), _{0} );'.\
                 format(self.cxxname)
-        else:
-            return 'set{0}( {1}( ::zerobuf::fromJSON< {2} >( ::zerobuf::getJSONField( json, "{3}" ))));'.\
+        # convert enums to their name as a string
+        if self.value_type.is_enum_type:
+            return 'set{0}( {1}( from_string( ::zerobuf::fromJSON< std::string >( ::zerobuf::getJSONField( json, "{3}" )))));'.\
+                format(self.cxxName, self.value_type.type, self.value_type.get_data_type(), self.cxxname)
```

Comment:
The format string doesn't have any {2}, but you are passing four parameters, review.